lcls-twincat-common-components/Library/Devices/PPM/FB_PPM_PowerMeter.TcPOU

```diff
@@ -99,19 +160,91 @@ fbGetPMVoltage(
     fTermMin := 0,
     fReal => fVoltage);
 
-// Power meter calibration
-fCalibBase := (fVoltage + fCalibRelOffset) * fCalibRelRatio;
-fCalibMJ := fCalibBase * fCalibMJRatio;
+IF R_TRIG(CLK:=(eEnumGet <> E_PPM_States.OUT OR stYAxisState <> 0)) THEN
```

Comment:
Thanks for catching the R_TRIG misuse. The logic is correct. When we are no longer stationary and at the OUT state, dump five seconds worth of background voltages, and set background variables to whatever they are expected when background measurements restart. This way it will automatically stop and start, with background collected even if in manual mode. Also, by resetting udBackgroundVoltageBufferIndex, we don't have to clear the array